service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationQueries.kt

```diff
@@ -95,9 +95,17 @@ fun Database.Transaction.clearOldReservations(
     return batch.executeAndReturn().mapTo<AttendanceReservationId>().toList()
 }
 
-fun Database.Transaction.clearReservationsForRange(childId: ChildId, range: DateRange): Int {
+fun Database.Transaction.clearReservationsForRangeExceptInHolidayPeriod(
+    childId: ChildId,
+    range: DateRange
+): Int {
     return this.createUpdate(
-            "DELETE FROM attendance_reservation WHERE child_id = :childId AND between_start_and_end(:range, date)"
+            """
+            DELETE FROM attendance_reservation
+            WHERE child_id = :childId
+            AND between_start_and_end(:range, date)
+            AND NOT EXISTS (SELECT 1 FROM holiday_period hp WHERE period @> date)
```

Comment:
nitpick: i'd like to see aliases, now period @> date is ambiguous unless you remember which of them is part of which table

Reply:
I had them first in this exact spot, but it looked so ugly 😄